### PR TITLE
Iox #408 rename publisher and subscriber

### PR DESCRIPTION
**Pull request by elBoberido**
## Pre-Review Checklist for the PR Author

1. [x] Branch follows the naming format (`iox-#123-this-is-a-branch`)
1. [x] Commits messages are according to this [guideline][commit-guidelines]
    - [x] Commit messages have the issue ID (`iox-#123 commit text`)
    - [x] Commit messages are signed (`git commit -s`)
    - [x] Commit author matches [Eclipse Contributor Agreement][eca] (and ECA is signed)
1. [x] Update the PR title
   - Follow the same conventions as for commit messages
   - Link to the relevant issue
1. [x] Relevant issues are linked
1. [x] Add sensible notes for the reviewer
1. [x] All checks have passed (except `task-list-completed`)
1. [x] Assign PR to reviewer

[commit-guidelines]: https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html
[eca]: http://www.eclipse.org/legal/ECA.php

## Notes for Reviewer
This PR renames the the `TypedPublisher` and `TypedSubscriber` to just `Publisher` and `Subscriber` as decided in #408. Renaming the untyped API will be done in a separate PR if it is still desired.

The `typed_publisher.hpp` and `typed_subscirber.hpp` were renamed with `git mv` but since there was already a `publisher.hpp` and `subscriber.hpp` git assumes those old files where changed.

## Checklist for the PR Reviewer

- [x] Commits are properly organized and messages are according to the guideline
- [x] Code according to our coding style and naming conventions
- [x] Unit tests have been written for new behavior
- [x] Public API changes are documented via doxygen
- [x] Copyright owner are updated in the changed files
- [x] PR title describes the changes

## Post-review Checklist for the PR Author

1. [x] All open points are addressed and tracked via issues

## Post-review Checklist for the Eclipse Committer

1. [x] All checkboxes in the PR checklist are checked or crossed out
1. [x] Merge

## References

- Partly closes **408**

### CODE REVIEW

File: iceoryx_examples/icedelivery/CMakeLists.txt [hunk 1]
```diff
@@ -27,11 +27,11 @@ if ( NOT ICEORYX_CXX_STANDARD )
     include(IceoryxPlatform)
```
Comment (elfenpiff) [inline iceoryx_examples/icedelivery/CMakeLists.txt:27]: Please update the copyright header.

File: iceoryx_examples/icedelivery/iox_subscriber_with_history.cpp [hunk 2]
```diff
@@ -16,7 +16,7 @@
 
 #include "topic_data.hpp"
```
Comment (elfenpiff) [inline iceoryx_examples/icedelivery/iox_subscriber_with_history.cpp:18]: Please update the copyright header.

File: iceoryx_examples/waitset/ice_waitset_individual.cpp [hunk 3]
```diff
@@ -14,7 +14,7 @@
 //
 // SPDX-License-Identifier: Apache-2.0
 
-#include "iceoryx_posh/popo/typed_subscriber.hpp"
```
Comment (elfenpiff) [inline iceoryx_examples/waitset/ice_waitset_individual.cpp:17]: Please update the copyright header.

File: iceoryx_posh/test/moduletests/test_popo_sample.cpp [hunk 4]
```diff
@@ -14,8 +14,8 @@
 //
```
Comment (elfenpiff) [inline iceoryx_posh/test/moduletests/test_popo_sample.cpp:14]: Please update the copyright header.

File: iceoryx_posh/test/moduletests/test_popo_base_subscriber.cpp [hunk 5]
```diff
@@ -31,10 +31,15 @@
 using namespace ::testing;
 using ::testing::_;
 
+// anonymous namespace to prevent linker issues or sanitizer false positives
+// if a struct with the same name is used in other tests
+namespace
+{
 struct DummyData
 {
     uint64_t val = 42;
```
Comment (dkroenke) [inline iceoryx_posh/test/moduletests/test_popo_base_subscriber.cpp:40]: :eyes:

File: iceoryx_posh/test/moduletests/test_popo_sample.cpp [hunk 6]
```diff
@@ -14,19 +15,24 @@
...
 
+// anonymous namespace to prevent linker issues or sanitizer false positives
+// if a struct with the same name is used in other tests
+namespace
+{
 struct DummyData
 {
     DummyData() = default;
-    int val = 42;
+    uint32_t val = 42;
```
Comment (elfenpiff) [inline iceoryx_posh/test/moduletests/test_popo_sample.cpp:33]: :octopus:  <- @dkroenke

**Review by elfenpiff — DISMISSED** [review]

File: iceoryx_examples/ice_multi_publisher/ice_subscriber.cpp [hunk 7]
```diff
@@ -14,7 +14,7 @@
 //
 // SPDX-License-Identifier: Apache-2.0
 
-#include "iceoryx_posh/popo/typed_subscriber.hpp"
```
Comment (mossmaurice) [inline iceoryx_examples/ice_multi_publisher/ice_subscriber.cpp:17]: s/2020, 2021/2020 - 2021

File: iceoryx_posh/test/moduletests/test_popo_sample.cpp (hunk 6 above)
Comment (elBoberido) [reply iceoryx_posh/test/moduletests/test_popo_sample.cpp:33]: btw. that was the reason for the address sanitizer issue. There were 4 structs with the same name, three with uint64_t and one with int all in the same binary. Somehow the address sanitizer mixed them up.

File: iceoryx_examples/ice_multi_publisher/ice_subscriber.cpp (hunk 7 above)
Comment (elBoberido) [reply iceoryx_examples/ice_multi_publisher/ice_subscriber.cpp:17]: I hate this copyright stuff 😄

File: iceoryx_posh/test/moduletests/test_popo_base_subscriber.cpp [hunk 8]
```diff
@@ -31,10 +31,15 @@
 using namespace ::testing;
 using ::testing::_;
 
+// anonymous namespace to prevent linker issues or sanitizer false positives
```
Comment (MatthiasKillat) [inline iceoryx_posh/test/moduletests/test_popo_base_subscriber.cpp:34]: I was aware of this and always have all my tests in a local anonymous namespace. This should be in the test recommendation file as well, if if is not already mentioned there.

Avoid weird linker issues, some of them seemingly silent...

File: iceoryx_posh/test/moduletests/test_popo_untyped_subscriber.cpp [hunk 9]
```diff
@@ -24,10 +24,15 @@
...
 
+// anonymous namespace to prevent linker issues or sanitizer false positives
+// if a struct with the same name is used in other tests
+namespace
+{
 struct DummyData
 {
     uint64_t val = 42;
 };
+} // namespace
```
Comment (MatthiasKillat) [inline iceoryx_posh/test/moduletests/test_popo_untyped_subscriber.cpp:35]: I would extend this to the test fixture or even the tests themselves. Test fixture names can also cause conflicts.

**Review by MatthiasKillat — APPROVED** [review]

**Review by mossmaurice — APPROVED** [review]

**Comment by MatthiasKillat** [comment]
I would recommend adding or extending anonymous namespaces to all tests in a separate PR to avoid future linker/address sanitizer issues. The should cover all definitions in the corresponding test_xyz.cpp file.

File: iceoryx_posh/test/moduletests/test_popo_untyped_subscriber.cpp (hunk 9 above)
Comment (elBoberido) [reply iceoryx_posh/test/moduletests/test_popo_untyped_subscriber.cpp:35]: addressed in #563

**Pull request merged**